### DISCUSSION TITLE
Workflow only executing on force push

### DISCUSSION THREAD

**Discussion by HusamElbashir** [community/community#25882] [Actions]
<p>I have <a href="https://github.com/HusamIbrahim/vue-ish/blob/dev/.github/workflows/main.yml" rel="nofollow noopener">this workflow</a> set up to execute on push. The weird thing is that it doesn’t execute when I push to the dev branch but whenever I force push it does! Why is that?</p>

**Comment by timruffles** [comment]
<p>Hi <a class="mention" href="/husamibrahim">@husamibrahim</a> , thanks for the repo. I can see the workflow has run on lots of commits on the <code>dev</code> branch. Please could you identify the commits at the head of the pushes you felt should have run but didn’t? Thanks!</p>

**Comment by HusamElbashir** [comment]
<p>Thanks <a class="mention" href="/timruffles">@timruffles</a> for taking the time to look into this. Here are a couple of commits which should have triggered the workflow but didn’t …</p>
<ul>
<li><a href="https://github.com/HusamIbrahim/vue-ish/commit/9722a292144baed3a598a72f72f002497b3e334d" rel="nofollow noopener">9722a292144baed3a598a72f72f002497b3e334d</a></li>
<li><a href="https://github.com/HusamIbrahim/vue-ish/commit/c3cd40a85e18963b521e819646caebdcf7d4cdf6" rel="nofollow noopener">c3cd40a85e18963b521e819646caebdcf7d4cdf6</a></li>
</ul>
<p>I didn’t bother force pushing in these two cases because these were minor commits, but they should have triggered the workflow according to the defined rules. Some of the other commits that did trigger the workflow such as this one (<a href="https://github.com/HusamIbrahim/vue-ish/commit/26b72ada308da7667119bbb97fa6973d57335ca5" rel="nofollow noopener">26b72ada308da7667119bbb97fa6973d57335ca5</a>) I pushed first and when the workflow didn’t run I kind of cheated by changing the commit hash at the head (git commit --amend --allow-empty) and then force pushing to get the workflow to run.</p>

**Comment by timruffles** [comment, marked as answer]
<p><a class="mention" href="/husamibrahim">@husamibrahim</a> yes, having a look that did indeed get incorrectly filtered out!</p>
<p>We’ve shipped a <a href="https://github.blog/changelog/2019-09-30-github-actions-event-filtering-updates/" rel="nofollow noopener">big update </a>to how filters work, so you shouldn’t see the same issue recur. If you do, please let me know :bowing_man:‍♂</p>
